deployer/src/project/service/state/d_started.rs

```diff
@@ -0,0 +1,181 @@
+    pub container: ContainerInspectResponse,
+    pub service: Option<Service>,
+    // Use default for backward compatibility. Can be removed when all projects in the DB have this property set
+    #[serde(default)]
+    pub stats: VecDeque<Stats>,
+}
+
+impl ServiceStarted {
+    pub fn new(container: ContainerInspectResponse, stats: VecDeque<Stats>) -> Self {
+        Self {
+            container,
+            service: None,
+            stats,
+        }
+    }
+}
+
+impl StateVariant for ServiceStarted {
+    fn name() -> String {
+        "Started".to_string()
+    }
+
+    fn as_state_variant(&self) -> String {
+        Self::name()
+    }
+}
+
+#[async_trait]
+impl<Ctx> State<Ctx> for ServiceStarted
+where
+    Ctx: DockerContext,
+{
+    type Next = ServiceReadying;
+    type Error = ServiceErrored;
+
+    #[instrument(skip_all)]
+    async fn next(self, ctx: &Ctx) -> Result<Self::Next, Self::Error> {
+        let Self {
+            container,
+            service,
+            mut stats,
+        } = self;
+        let container = container.refresh(ctx).await?;
+        let service = match service {
+            Some(service) => service,
+            None => Service::from_container(container.clone())?,
+        };
+
+        let service_id = Ulid::from_string(service.id.as_str())
+            .map_err(|_| ServiceErrored::internal("failed to get the service id"))?;
+
+        let mut runtime_client = ctx
+            .runtime_manager()
+            .runtime_client(service_id, service.target)
+            .await
+            .expect("to create a runtime client");
+        if runtime_client.health_check(Ping {}).await.is_ok() {
+            debug!("the service runtime responded to health check");
+            let idle_minutes = container.idle_minutes();
+
+            // Idle minutes of `0` means it is disabled and the project will always stay up
+            if idle_minutes < 1 {
+                Ok(Self::Next::Ready(ServiceReady {
+                    container,
+                    service,
+                    stats,
+                }))
+            } else {
+                let new_stat = ctx
+                    .docker()
+                    .stats(
+                        safe_unwrap!(container.id),
+                        Some(StatsOptions {
+                            one_shot: true,
+                            stream: false,
+                        }),
+                    )
+                    .next()
+                    .await
+                    .unwrap()?;
+
+                stats.push_back(new_stat.clone());
+
+                let mut last = None;
+
+                while stats.len() > (idle_minutes as usize) {
+                    last = stats.pop_front();
+                }
+
+                if let Some(last) = last {
+                    let cpu_per_minute = (new_stat.cpu_stats.cpu_usage.total_usage
+                        - last.cpu_stats.cpu_usage.total_usage)
+                        / idle_minutes;
```

Comment:
This is where we do the idle check everything it is triggered by ambulance / refresh